src/geo/projection/globe.js

```diff
@@ -288,11 +288,9 @@ export function globeBuffersForTileMesh(painter: Painter, tile: Tile, coord: Ove
     return [gridBuffer, poleBuffer];
 }
 
-export function globeMatrixForTile(id: CanonicalTileID, globeMatrix: mat4) {
+export function globeMatrixForTile(id: CanonicalTileID, globeMatrix: Float64Array) {
     const decode = globeDenormalizeECEF(globeTileBounds(id));
-    const posMatrix = mat4.copy(new Float64Array(16), globeMatrix);
-    mat4.mul(posMatrix, posMatrix, decode);
-    return posMatrix;
+    return mat4.mul(mat4.create(), globeMatrix, decode);
```

Comment:
Is there a functional change from this diff, switching to `Float32` due to the new call to `create()`?

`mat4.create()`:
```
function create() {
  var out = new glMatrix.ARRAY_TYPE(16);
```
where `ARRAY_TYPE` is the following:
```
var ARRAY_TYPE = typeof Float32Array !== 'undefined' ? Float32Array : Array;
```

Reply:
@karimnaaji yes, this is intentional since we directly pass the returned matrix to `uniformMatrix` later. Multiplication itself remains precise though (`globeMatrix` and `decode` are double precision).